docs/symbols.md

```diff
@@ -426,16 +428,9 @@ Put another way, string-to-SID mappings have the following precedence:
 System Symbols
 --------------
 
-The version included in the system identifier is independent of the
-version of the implied system symbol table (named `"$ion"`). Each version of
-the Ion specification defines the corresponding system symbol table version.
-Ion 1.0 uses the `"$ion"` symbol table, version 1, and future versions of Ion
-will use larger versions of the `"$ion"` symbol table. `$ion_1_1` will probably
-use version 2, while `$ion_2_0` might use version 5.
-
-Applications and users should never have to care about these symbol
-table versions, since they are never explicit in user data: this
-specification disallows (by ignoring) imports named `"$ion"`.
+Each version of the Ion specification defines a corresponding system symbol
+table. Applications and users should never have to care about these system
+symbol tables, since they are never explicit in user data.
 
```

Comment:
This change is significant, and I'm overall not inclined to make it without good reason.  The original text was intended to ensure that system symbols are not removed nor renumbered nor reassigned, because doing so adds unnecessary complexity to the system-level processing.  It puts the system symbol table compatibility mechanisms on the same ground as shared symbol tables, and really just treats the system table as a built-in shared table that's automatically imported.

Reply:
After this change, system symbol tables can still be treated as built-in shared symbol tables that are automatically imported. The change just clarifies that system symbol tables do not need to relate to each other across versions.

I don't think I agree that this adds significant complexity. Implementations already will need to swap in logic upon encountering an IVM that declares a different Ion version. Part of this logic would be to use the correct system symbol table.

Reply:
I don't think we should change the definition for Ion 1.0, it is what it is and I don't think we should back-track on it.  

@toddjonker - can you help elucidate why Ion 2.0 would be required to have a version bump on the `$ion` symbol table (which is already this weird thing that no one can reference).  I am a little more amenable to Ion 1.x requiring support of the of previous symbols to keep the implementations simpler.

Another way I can think of this proposed change, is that the system symbol table is uniquely identified by the Ion version and has no user defined name (i.e. you cannot import it).  I don't really find this onerous to implement personally, but I grant that it adds some marginal complexity.

Reply:
> elucidate why Ion 2.0 would be required to have a version bump on the $ion symbol table

I don't intend to say that (today).  I'm saying that, today, we have no idea what Ion 2.0 needs, and therefore this change is premature.  With 2.0 we could throw all of this out if we have better ideas and can provide a smooth migration path.

I also don't see what benefit we get by retroactively dropping the spec that system symbol tables have a well-defined name.  The intent is that system symbol tables _are_ shared symbol tables, and shared symtabs have names and versions.  This is already encoded in our APIs:

> `SymbolTable.isSystemTable()`
> Determines whether this symbol table is a system symbol table, and therefore shared, named, versioned, and read-only.

What benefit is achieved by allowing implementations to have different names for the same symtab?